neurots/extract_input/from_neurom.py

```
@@ -14,92 +14,178 @@
+
+    def _prob(_angle):
+        return prob_function(
+            _angle, [scale_low, rate_low, scale_high, rate_high], form="double_step"
+        ) * np.sin(_angle)
+
+    return _prob(angle) / quad(_prob, 0, np.pi)[0]
+
+
+def get_fit_prob_function(morph_class="PC", neurite_type=nm.BASAL_DENDRITE, params=None):
+    """Get probability function for trunk angles.
+
+    Args:
+        morph_class (str): morphological class (PC or IN)
+        neurite_type (nm.NeuriteType): type of neurite to consider
+        params (dict): is not None, can be used to overwrite the fit functions/bounds
     """
-    if opt_distr.type == "norm":
-        return {"norm": {"mean": opt_distr.params[0], "std": opt_distr.params[1]}}
-    elif opt_distr.type == "uniform":
-        return {
-            "uniform": {
-                "min": opt_distr.params[0],
-                "max": opt_distr.params[1] + opt_distr.params[0],
-            }
-        }
-    elif opt_distr.type == "expon":
-        return {"expon": {"loc": opt_distr.params[0], "lambda": 1.0 / opt_distr.params[1]}}
-    return None
+    # this is the form of the expected parameter dict.
+    # The bounds correspond to the function's parameter min/max bounds for the fit
+    default_params = {
+        "PC": {
+            "basal_dendrite": {"form": "step", "bounds": ([0.0, 0.01], [np.pi, 10])},
+            "apical_dendrite": {"form": "step", "bounds": ([0.0, -10], [3.0, -0.010])},
+            "axon": {"form": "step", "bounds": ([0.0, 0.01], [3.0, 10])},
+        },
+        "IN": {
+            "basal_dendrite": {
+                "form": "double_step",
+                "bounds": ([-np.pi, 0.01, -np.pi, 0.01], [np.pi, 10, np.pi, 10]),
+            },
+            "axon": {"form": "step", "bounds": ([0.0, 0.01], [3.0, 10])},
+        },
+    }
+    if params is not None:
+        default_params.update(params)
 
+    bound = default_params[morph_class][neurite_type.name]["bounds"]
```

Maybe we should raise a clear exception if `morph_class` is not in `["PC", "IN"]`, because here it will just return a `KeyError` which might not be clear for the user.